quarkus/commons/src/main/java/org/apache/polaris/commons/PostgresRelationalJdbcLifeCycleManagement.java

```diff
@@ -53,19 +60,42 @@ public Map<String, String> start() {
 
     context.containerNetworkId().ifPresent(postgres::withNetworkMode);
     postgres.start();
-    return Map.of(
-        "polaris.persistence.type",
-        "relational-jdbc",
-        "quarkus.datasource.db-kind",
-        "pgsql",
-        "quarkus.datasource.jdbc.url",
-        postgres.getJdbcUrl(),
-        "quarkus.datasource.username",
-        postgres.getUsername(),
-        "quarkus.datasource.password",
-        postgres.getPassword(),
-        "quarkus.datasource.jdbc.initial-size",
-        "10");
+    Map<String, String> props = new HashMap<>();
+    props.put("polaris.persistence.type", "relational-jdbc");
+
+    if (!databases.isEmpty()) {
+      for (String database : databases) {
+        // polaris.relation.jdbc.datasource.realm=realm_ds
+        props.put(String.format("polaris.relation.jdbc.datasource.%s", database), database + "_ds");
+        props.put(String.format("quarkus.datasource.%s.db-kind", database + "_ds"), "pgsql");
+        props.put(String.format("quarkus.datasource.%s.active", database + "_ds"), "true");
+        props.put(
+            String.format("quarkus.datasource.%s.jdbc.url", database + "_ds"),
+            postgres.getJdbcUrl().replace("realm1", database));
+        props.put(
+            String.format("quarkus.datasource.%s.username", database + "_ds"),
+            postgres.getUsername());
+        props.put(
+            String.format("quarkus.datasource.%s.password", database + "_ds"),
+            postgres.getPassword());
+      }
+    } else {
+      return Map.of(
+          "polaris.persistence.type",
+          "relational-jdbc",
+          "quarkus.datasource.db-kind",
+          "pgsql",
+          "quarkus.datasource.jdbc.url",
+          postgres.getJdbcUrl(),
+          "quarkus.datasource.username",
+          postgres.getUsername(),
+          "quarkus.datasource.password",
+          postgres.getPassword(),
+          "quarkus.datasource.jdbc.initial-size",
+          "10");
+    }
+
+    return props;
```

Comment:
The code patterns of falling through to here from the main `if` branch, but not doing so from `else` look very odd to me from code maintenance POV.